packages/api-docs-builder/ApiBuilders/ComponentApiBuilder.ts

```diff
@@ -628,58 +628,59 @@ export default async function generateComponentApi(
   const filename = componentInfo.filename;
   let reactApi: ComponentReactApi;
 
-  if (componentInfo.isSystemComponent || componentInfo.name === 'Grid2') {
```

Comment:
That hard-coded logic felt wrong, we further added stuff around it in https://github.com/oliviertassinari/material-ui/commit/08dce48d69c0cda46455d5ce35a329bb7f6f6edc#diff-eef575d465fe9894710e7eb399c76ad3e3900a8cbb6d374be7f303c4b6c6ef51R627.